deliver/lib/deliver/upload_metadata.rb

```diff
@@ -226,30 +226,47 @@ def verify_available_languages!(options)
       # Collect all languages we need
       # We only care about languages from user provided values
       # as the other languages are on iTC already anyway
-      v = options[:app].edit_version
-      UI.user_error!("Could not find a version to edit for app '#{options[:app].name}', the app metadata is read-only currently") unless v
+      v = options[:app].details
+      UI.user_error!("Could not find details for app '#{options[:app].name}'") unless v
+
+      # find languages currently enabled on ITC (enabled_languages)
+      enabled_languages = {}  # map of locale code to language data hash
+      primary_data = nil      # the primary language data, e.g. english name/subtitle
```

Comment:
`primary_language` or `main_language` maybe?
(or `primary_language_data`)

Reply:
i think primary data is a hash of information representing the settings for the primary language/locale